api: rename Cluster kind to ScyllaCluster in missing places

zimnx opened the pull request:
Problems solved:
 - webhooks were watching old "Cluster" resource
- controller wasn't able to update cluster status
- controller watch on owned StatefulSet events was registered on old "Cluster" resource
    
Fixes #144

espindola commented:
Please open one issue for adding regression tests for each of the problems you found.

Some of the files in here are generated, no? Please split the patch in

* Change the non generated files
* Change to the generated files

In the second patch, say which command was used to regenerate them.

zimnx commented:
> Please open one issue for adding regression tests for each of the problems you found.
> 
Added issue to implement basic scenario first. I wasn't able to spawn any node using master due to above problems. https://github.com/scylladb/scylla-operator/issues/220

> Some of the files in here are generated, no? Please split the patch in
> 
>     * Change the non generated files
> 
>     * Change to the generated files
> 
> 
> In the second patch, say which command was used to regenerate them.

done

espindola commented on examples/eks/cluster.yaml:
```diff
@@ -40,7 +40,7 @@ rules:
       - scylla.scylladb.com
     resources:
-      - clusters
+      - scyllaclusters
```
This file is a copy and paste from the one in config, no? It seems to have some arbitrary dependencies:

````
$ diff -u  ./config/manager/cluster/cluster.yaml examples/eks/cluster.yaml
... 
apiVersion: scylla.scylladb.com/v1alpha1
 kind: Cluster
 metadata:
-  name: cluster
-  namespace: scylla-manager
+  name: scylla-cluster
+  namespace: scylla
```

zimnx replied:
`cluster.yaml` files are not generated. These are examples what user should provide and these were manually crafted or copied from generic if they suits.

espindola commented on examples/generic/cluster.yaml:
```diff
@@ -40,7 +40,7 @@ rules:
       - scylla.scylladb.com
     resources:
-      - clusters
+      - scyllaclusters
```
and this one.

espindola commented on examples/gke/cluster.yaml:
```diff
@@ -40,7 +40,7 @@ rules:
       - scylla.scylladb.com
     resources:
-      - clusters
+      - scyllaclusters
```
and this one.

espindola commented on pkg/api/v1alpha1/zz_generated.deepcopy.go:
```diff
@@ -21,7 +21,7 @@ limitations under the License.
 
 import (
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
According to the file name this file is generated.

zimnx commented:
@espindola I found that autogeneration stopped working some time ago or wasn't there in some files. I fixed the issues, extracted to separate commits. PTAL

espindola commented on pkg/api/v1alpha1/cluster_webhook.go:
```diff
@@ -33,14 +33,11 @@ func (r *ScyllaCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Is this change part of running "make manifests"?

zimnx replied:
no, i manually removed this comment. It was generated by the `kubebuilder` during project initializaiton.

espindola commented on config/manager/rbac/role.yaml:
```diff
@@ -6,19 +6,25 @@ metadata:
   creationTimestamp: null
   name: manager-role
 rules:
-- resources:
```
This file is modified in 3 of the 4 patches. It is auto generated, correct? If so, it should be modified in only one patch, no?

zimnx replied:
done, somehow missed it during rebasing and squashing

espindola reviewed (commented):
Thanks a lot for fixing the Makefile.

I think this needs to be rebased on top of current master.

Auto generated files are still modified in multiple patches. Why?

espindola reviewed (approved):
LGTM
